customer.go

```diff
@@ -17,75 +19,31 @@ type Customer struct {
 }
 
 // Create a customer
-func CreateCustomer(client *Client, data *Customer) {
-	values := map[string]interface{}{
-		"auth":         client.Auth,
-        "api_key":      client.APIKey,
-        "firstname":    data.Firstname,
-        "lastname":     data.Lastname,
-        "email":        data.Email,
-        "phone":        data.Phone,
-        "user_id":      data.UserId,
-        "notes":        data.Notes,
-        "location_id":  data.LocationId,
-	}
-
+func CreateCustomer(data *Customer) map[string]interface{} {
     endpoint := APIBaseUrl + "customers/create"
-
-    Response(values, endpoint)
+    return Response(data, endpoint)
 }
 
 // Retrieve a list of customers
-func RetrieveCustomers(client *Client) {
-    values := map[string]interface{}{
-		"auth": client.Auth,
-        "api_key": client.APIKey,
-    }
-
+func RetrieveCustomers(data *Customer) map[string]interface{} {
     endpoint := APIBaseUrl + "customers"
-
-    Response(values, endpoint)
+    return Response(data, endpoint)
 }
 
 // Retrieve a single customer
-func RetrieveCustomer(client *Client, data *Customer) {
-    values := map[string]interface{}{
-		"auth": client.Auth,
-        "api_key": client.APIKey,
-    }
-
+func RetrieveCustomer(data *Customer) map[string]interface{} {
```

Comment:
More of a comment on the application structure, but Customer models shouldn't be available in sequential numerical order. Referencing by UUID instead (https://laravel-news.com/eloquent-uuid) externally & internally prevents data leakage from the data structure.
e.g. Being able to tell exactly how many customers are currently on the platform by generating an account and checking the ID.

Reply:
I totally agree! This is ultimately a shortcoming of the API itself - something I plan to eventually implement is public/private ID's so that we aren't exposing those internal identifiers.